test/test_sync.py

```diff
@@ -105,6 +107,58 @@ def test_invalid_permissions(self):
                 os.path.join(tmpdir, self.NAMES[0])
             )
 
+    def test_prefix_calculation(self):
+        """
+
+            https://github.com/Backblaze/B2_Command_Line_Tool/issues/334
+
+            This test case attempts to stress the prefix calculation for the `all_files`
+                generator in the LocalFolder implementation of AbstractFolder.
+
+            Paths can be generalized to the following cases for linux:
+                  1) `/`
+                  2) `/{m}/`
+                  3) `/{m}`
+                  4) `{m}`
+                  5) `{m}`
+            Where except for case 1), `N = len({m}) + 1` and `N` is the necessary
+                prefix length to yield an empty string when taking the list difference
+                of the absolute and relative paths.
+
+            For windows, case 1) is different (e.g. C:\\) but the others should be the same.
+
+        """
+
+        prefix = os.path.normpath(u'C://') if platform.system() == 'Windows' else unicode(os.path.sep)
+        separator = unicode(os.path.sep)
+        root = os.path.normpath(os.getcwdu())  # Always has leading but not trailing separator.
+        relative = u''.join(choice(lowercase) for _ in range(randint(4, 8)))
+        permutations = dict(**{
```

Comment:
How about just `{ ... }` instead of `dict(**{ ... })`?

Reply:
Yep sorry, got fooled by my IDE there.